Add helm chart

antoniolago opened the pull request:
As discussed in #55, users should have helm chart as option for deploying headplane, I forked an existing chart and fixed a few things.
I will keep https://github.com/antoniolago/headplane-chart/ as is, but the mantainers may want the helm's code into this repo tree so other devs can use and enhance it in a more official way.

I used my personal OCI registry on README, feel free to change this or any other thing.

I have deployed and tested >some< features and it worked (basic admin access, add device, edit acl via values.yaml) so it definetly should be more tested.

Edit:
Using this chart at a production environment and everything looks good

antoniolago committed:
Add helm chart

antoniolago committed:
Remove default annotation for ingress

that wasnt supposed to be there

antoniolago committed:
Fix static namespace + Fix relay configs + secrets overhaul + helm-docs

antoniolago committed:
oidc config + configmap/secrets overhaul

antoniolago committed:
fix oidc docs

tale commented:
Whenever you're ready for review and merge, feel free to change it off of a draft and I'll take a look. Take your time though, we have a long time to go before 0.6.1!

antoniolago committed:
Add annotations/labels to pvc

antoniolago committed:
minor doc adjust

antoniolago committed:
Restore acl value

This is only applicable if the user choose policy mode file, but without this value the configmap is not being created

antoniolago committed:
add null check cookiesecret

Co-Authored-By: definitelynobody <51100439+definitelynobody@users.noreply.github.com>

tale commented:
Thank you for all the hard work, I'll try to get it merged into 0.6.1 ASAP.

antoniolago committed:
Simplify PVC structure on manifests and update headscale version

Co-Authored-By: definitelynobody <51100439+definitelynobody@users.noreply.github.com>

tale reviewed (changes requested):
Aside from nits related to the packaging and distribution of the charts, this looks good to me.

antoniolago committed:
Add headscale.oidc.pkce

Co-Authored-By: Stephan Deumier <github@deumier.dev>

antoniolago committed:
Fix oidc missing configs + test cases

Co-Authored-By: Stephan Deumier <github@deumier.dev>

antoniolago commented on helm/README.md:
````diff
@@ -0,0 +1,178 @@
+### Install the Chart
+```sh
+# Install with default values
+helm install headplane oci://harbor.lag0.com.br/library/headplane
````
I use Harbor just because I like self-hosting stuff lol, no objections for creating a workflow to push this to another chart repository, maybe you can setup a workflow based on this one: https://github.com/antoniolago/headplane-chart/blob/main/.github/workflows/publish-helm-chart.yml (tests are included in this PR already)

tale commented on helm/README.md:
```diff
@@ -0,0 +1,178 @@
+Consider using a secrets management solution in production like external-secrets.
+
+## License
+Copyright © 2025 antoniolago
```
Then in that case, we do need their permission yes, and I can't technically merge this in-tree until it's been addressed.

antoniolago commented:
> @antoniolago @tale - I have been using this helm chart locally but saw you were contributing a fixed version of the existing one so just kept it to myself.
> 
> #263
> 
> If licensing is a concern, I am happy to provide that under MIT as an option - no hard feelings at all if this one works out and you decide to close my PR
> 
> If you find it helpful, @antoniolago I would love to collaborate with you on bringing this up to standard

Hello, nice work you're doing, havent got the opportunity to test yet but it looks like you know what you are doing, at the moment I've reached a goal state to >my specific requirements< with headplane's infra, this PR and the chart repository is a way to share this but it's not necessarely the best chart option, tho we could ask the 4 people involved if they agree to change license theres a certainty on your solution to the license issue, so maybe we should actually go for it! At the moment I'm changing obsessions lol, but I'll keep an eye on this for sure.

lucasfcnunes reviewed (commented):
*multiple files have missing new line at EOF
